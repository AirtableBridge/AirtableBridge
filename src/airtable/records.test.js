const records = require("./records");

function getFix(name) {
  return require(`../fixtures/${name}.json`);
}

function formatCalls(api) {
  const str = getCalls(api).map(
    ({ method, calls }) =>
      `${method}\n${calls
        .map(call => call.map(foo => JSON.stringify(foo, null, 2)).join("\n"))
        .join("\n")}`
  );
  return str.join("\n\n");
}

function getCalls(api) {
  return Object.keys(api)
    .map(method => ({ method, calls: api[method].mock.calls }))
    .filter(({ calls }) => calls.length > 0);
}

function createMock() {
  const mocks = {
    create: jest.fn().mockReturnValue(Promise.resolve({ yo: 2 })),
    update: jest.fn().mockReturnValue(Promise.resolve()),
    get: jest.fn().mockReturnValue(Promise.resolve())
  };

  const airtable = () => mocks;

  return { mocks, airtable };
}

describe("records", () => {
  it("create issue", async () => {
    const { airtable, mocks } = createMock();
    await records.create(airtable, "Issues", getFix("issue").issue);

    // console.log(formatCalls(mocks));
    expect(formatCalls(mocks)).toMatchSnapshot();
  });

  it("update issue", async () => {
    const { airtable, mocks } = createMock();
    await records.update(airtable, "Issues", getFix("issue"));

    // console.log(formatCalls(mocks));
    expect(formatCalls(mocks)).toMatchSnapshot();
  });

  it("create pr", async () => {
    const { airtable, mocks } = createMock();
    await records.create(airtable, "Pulls", getFix("pr"));

    // console.log(formatCalls(mocks));
    expect(formatCalls(mocks)).toMatchSnapshot();
  });
});
