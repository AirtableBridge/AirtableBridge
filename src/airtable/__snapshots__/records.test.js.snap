// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`records create issue 1`] = `
"create
{
  \\"Title\\": \\"Debugging Probot\\",
  \\"ID\\": 4384,
  \\"Open\\": true,
  \\"URL\\": \\"https://github.com/devtools-html/debugger.html/issues/4384\\",
  \\"Assignees\\": [
    \\"belld19233\\"
  ],
  \\"Description\\": \\"Ignore this bug. Instead of making comments on bugs in progress, I will do so here in order to debug the issue of Probot not detecting some /claim comments.\\\\r\\\\n\\\\r\\\\nAlso, this will test our Probot's ability to assign users to a bug.\\",
  \\"Labels\\": [
    \\"in progress\\"
  ],
  \\"Author\\": \\"belld19233\\",
  \\"Created\\": \\"10/16/2017\\",
  \\"Updated\\": \\"10/20/2017\\",
  \\"Closed\\": null
}"
`;

exports[`records create pr 1`] = `
"create
{
  \\"ID\\": 4478,
  \\"Url\\": \\"https://github.com/devtools-html/debugger.html/pull/4478\\",
  \\"Open\\": true,
  \\"Title\\": \\"Setup breakpoint dropdown\\",
  \\"Created\\": \\"10/23/2017\\",
  \\"Updated\\": \\"10/23/2017\\",
  \\"Closed\\": null,
  \\"Merged\\": null,
  \\"Description\\": \\"Associated Issue: #4410 \\\\r\\\\n\\\\r\\\\n### Summary of Changes\\\\r\\\\n\\\\r\\\\n* Created breakpoint dropdown styling \\\\r\\\\n* Breakpoint toggle returns dropdown\\\\r\\\\n* Changed dropdown .svg to \\\\\\"+\\\\\\"\\\\r\\\\n\\\\r\\\\nExample test plan:\\\\r\\\\n\\\\r\\\\n- [x] Create an arbitrary breakpoint\\\\r\\\\n- [x] Breakpoint window displays panel toggle \\\\r\\\\n- [x] Clicking “+” opens the panel\\\\r\\\\n\\\\r\\\\nTODO:\\\\r\\\\n* Adjust dropdown width to fit parent\\\\r\\\\n* Add menu items (with corresponding images)\\\\r\\\\n* Add menu item implementation\\\\r\\\\n* Remove CommandBar options as needed\\",
  \\"Author\\": \\"GarethSharpe\\",
  \\"Avatar\\": [
    {
      \\"url\\": \\"https://avatars1.githubusercontent.com/u/22062107?v=4\\"
    }
  ],
  \\"Branch\\": \\"GarethSharpe:master\\"
}"
`;

exports[`records update issue 1`] = `""`;
